pkg/controller/constraint/constraint_controller.go

```diff
@@ -323,10 +325,27 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 				if !isAPIEnabled {
 					r.log.V(1).Info("Warning: VAP API is not enabled, cannot create VAPBinding")
 					generateVAPB = false
-				}
-				if !HasVAPCel(ct) {
-					r.log.V(1).Info("Warning: ConstraintTemplate does not contain VAP-style CEL source, cannot create VAPBinding")
-					generateVAPB = false
+				} else {
+					unversionedCT := &templates.ConstraintTemplate{}
+					if err := r.scheme.Convert(ct, unversionedCT, nil); err != nil {
+						status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+						if err2 := r.writer.Update(ctx, status); err2 != nil {
+							log.Error(err2, "could not update constraint status error for converting convert ConstraintTemplate to unversioned")
+						}
+						return reconcile.Result{}, err
+					}
+					hasVAP, err := ShouldGenerateVAP(unversionedCT)
+					if err != nil {
+						status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+						if err2 := r.writer.Update(ctx, status); err2 != nil {
+							log.Error(err2, "could not update constraint status error when determining if CT should generate VAP")
+						}
+						generateVAPB = false
+					}
+					if !hasVAP {
+						r.log.V(1).Info("Warning: ConstraintTemplate will not create ValidatingAdmissionPolicy, cannot create VAPBinding")
```

Comment:
```suggestion
						r.log.V(1).Info("Warning: Conditions are not satisfied to generate ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding")
```

Reply:
Should this be part of status error? and is this also reported as part of constraint template status?

Reply:
This is not specifically an error right? reporting it as error might be confusing. 

We discussed following up with designing a shcema for `VAP/VAPB` related errors and warnings on constraints and constrainttemplates.

Reply:
ok lets leave it as a log for now for both constraint and constraint template. can you add this to the next meeting agenda so we can discuss?

Reply:
Added to the agenda for the next meeting.